src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIAttackRangedBowBarbarian.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    /**
+     * Resets the task
+     */
+    public void resetTask()
+    {
+        super.resetTask();
+        this.entity.setSwingingArms(false);
+        this.seeTime = 0;
+        this.attackTime = -1;
+        this.entity.resetActiveHand();
+    }
+
+    /**
+     * Updates the task
+     */
+    public void updateTask()
+    {
+        final EntityLivingBase entitylivingbase = this.entity.getAttackTarget();
+
+        if (entitylivingbase != null)
+        {
+            final double d0 = this.entity.getDistanceSq(entitylivingbase.posX, entitylivingbase.getEntityBoundingBox().minY, entitylivingbase.posZ);
+            final boolean flag = this.entity.getEntitySenses().canSee(entitylivingbase);
+            final boolean flag1 = this.seeTime > 0;
+
+            if (flag != flag1)
+            {
+                this.seeTime = 0;
+            }
+
+            if (flag)
+            {
+                ++this.seeTime;
+            }
+            else
+            {
+                --this.seeTime;
+            }
+
+            if (d0 <= (double) this.maxAttackDistance && this.seeTime >= 20)
+            {
+                this.entity.getNavigator().clearPathEntity();
+                ++this.strafingTime;
+            }
+            else
+            {
+                this.entity.getNavigator().tryMoveToEntityLiving(entitylivingbase, this.moveSpeedAmp);
+                this.strafingTime = -1;
+            }
+
+            final int strafeTime = 20;
+
+            if (this.strafingTime >= strafeTime)
+            {
+                final double maxRNG = 0.3D;
+
+                if ((double) this.entity.getRNG().nextFloat() < maxRNG)
+                {
+                    this.strafingClockwise = !this.strafingClockwise;
+                }
+
+                if ((double) this.entity.getRNG().nextFloat() < maxRNG)
+                {
+                    this.strafingBackwards = !this.strafingBackwards;
+                }
+
+                this.strafingTime = 0;
+            }
+
+            if (this.strafingTime > -1)
+            {
+                if (d0 > (double) (this.maxAttackDistance * 0.75F))
+                {
+                    this.strafingBackwards = false;
+                }
+                else if (d0 < (double) (this.maxAttackDistance * 0.25F))
+                {
+                    this.strafingBackwards = true;
+                }
+
+                this.entity.getMoveHelper().strafe(this.strafingBackwards ? -0.5F : 0.5F, this.strafingClockwise ? 0.5F : -0.5F);
+                this.entity.faceEntity(entitylivingbase, 30.0F, 30.0F);
+            }
+            else
+            {
+                this.entity.getLookHelper().setLookPositionWithEntity(entitylivingbase, 30.0F, 30.0F);
+            }
+
+            final int alternateAttackTime = --this.attackTime;
+
+            if (this.entity.isHandActive())
+            {
+                if (!flag && this.seeTime < -60)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)